src/allUniq.js

```diff
@@ -0,0 +1,31 @@
+var _curry1 = require('./internal/_curry1');
+var _indexOf = require('./internal/_indexOf');
+
+
+/**
+ * Returns `true` if all elements are unique, in `R.equals` terms,
+ * otherwise `false`.
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig [a] -> Boolean
+ * @param {Array} list The array to consider.
+ * @return {Boolean} `true` if all elements are unique, else `false`.
+ * @example
+ *
+ *      R.allUniq(['1', 1]); //=> true
+ *      R.allUniq([1, 1]);   //=> false
+ *      R.allUniq([[42], [42]]); //=> false
+ */
+module.exports = _curry1(function allUniq(list) {
+  var len = list.length;
+  var idx = 0;
+  while (idx < len) {
```

Comment:
It doesn't affect this case since we're not invoking any user supplied function in the loop. The problems happens when you invoke a function that might modify the array you're looping over.

I agree with @davidchambers that we should not cache the length here.